README.rst

```diff
@@ -124,15 +124,26 @@ It will create a file in your package named ``_version.py`` like this:
    __all__ = ["__version__"]
 
 
-Then, so users of your project can find your version, in your root package's ``__init__.py`` add:
+Subsequent installations of your project will then use Incremental for versioning.
+
+
+Runtime integration
+~~~~~~~~~~~~~~~~~~~
+
+You may expose the ``incremental.Version`` from ``_version.py`` in your package's API.
+To do so, add to your root package's ``__init__.py``:
 
 .. code:: python
 
    from ._version import __version__
 
+.. note::
 
-Subsequent installations of your project will then use Incremental for versioning.
+    Providing a ``__version__`` attribute is falling out of fashion following the introduction of `importlib.metadata.version() <https://docs.python.org/3/library/importlib.metadata.html#distribution-versions>`_ in Python 3.6, which can retrieve an installed package's version.
 
+If you don't expose this object publicly, nor make use of it within your package,
```

Comment:
Just asking ?

If you don't make use of `incremental` inside your package, what's the point of using `incremental` ?

I find that `incremental` is usefull to track `NEXT` version to make it easy to do relase candidates and autoamtically have the version updated in your codebase.

Reply:
You have it exactly right &mdash; in 2024 Incremental's value is:

1. Managing CalVer versions
2. Updating indeterminate versions within the codebase

I view the runtime API (`incremental.Version`) as a legacy feature. There's no need to remove or deprecate it (it's part of the public API of a bunch of Twisted packages, that would just cause churn) but I don't want to advise new users adopt it as a matter of course. `importlib.metadata` is a better solution.

Here I'm trying to rework the docs to emphasize the new way.

Reply:
I don't think there's anything wrong with not caring about `Version`, but does `importlib.metadata` even provide a version object somewhere? It looks like it just returns strings.

Reply:
I think this is a closer analogue: https://packaging.pypa.io/en/stable/version.html#packaging.version.Version.__init__

Reply:
Yeah, `importlib.metadata` is just strings AFAIK. There are use cases that only need strings, though, like displaying the versions of everything installed, so I think that's a reasonable way to scope it. If you want a `Version` of whatever flavor it's easy enough to parse the string.